commit ae06985a47a4d912919dd0c1259425f1ea1c9133
Author: Eirik S. Morland <eirik@morland.no>
Date:   Sun Nov 23 14:34:23 2025 +0100

    Avoid errors in PHP 8.5

diff --git a/src/RequirementChecker/RequirementsBuilder.php b/src/RequirementChecker/RequirementsBuilder.php
index 1cf8a933..77e558fe 100644
--- a/src/RequirementChecker/RequirementsBuilder.php
+++ b/src/RequirementChecker/RequirementsBuilder.php
@@ -58,6 +58,9 @@ final class RequirementsBuilder
 
     public function addRequiredExtension(Extension $extension, ?string $source): void
     {
+        if (!$source) {
+            $source = '';
+        }
         $this->requiredExtensions[$extension->name][] = $source;
         $this->allExtensions[$extension->name][$source] = [$source, RequirementType::EXTENSION];
     }
