From 77c0b71d0548ad17c416274ec18275e2eb26d7ea Mon Sep 17 00:00:00 2001
From: "Eirik S. Morland" <eirik@morland.no>
Date: Mon, 23 Sep 2024 10:23:26 +0200
Subject: [PATCH] Fix implicit nullable type to avoid PHP 8.4 warnings

---
 src/HttpMethodsClient.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/HttpMethodsClient.php b/src/HttpMethodsClient.php
index 497921b..95fee3b 100644
--- a/src/HttpMethodsClient.php
+++ b/src/HttpMethodsClient.php
@@ -33,7 +33,7 @@ final class HttpMethodsClient implements HttpMethodsClientInterface
     /**
      * @param RequestFactory|RequestFactoryInterface $requestFactory
      */
-    public function __construct(ClientInterface $httpClient, $requestFactory, StreamFactoryInterface $streamFactory = null)
+    public function __construct(ClientInterface $httpClient, $requestFactory, ?StreamFactoryInterface $streamFactory = null)
     {
         if (!$requestFactory instanceof RequestFactory && !$requestFactory instanceof RequestFactoryInterface) {
             throw new \TypeError(
